packages/core/usage-data/core-usage-data-server-internal/src/core_usage_stats_client.ts

```diff
@@ -76,17 +80,59 @@ const SPACE_CONTEXT_REGEX = /^\/s\/([a-z0-9_\-]+)/;
 
 /** @internal */
 export class CoreUsageStatsClient implements ICoreUsageStatsClient {
+  private readonly fieldsToIncrement$ = new Subject<string[]>();
+  private readonly flush$ = new Subject<void>();
+
   constructor(
     private readonly debugLogger: (message: string) => void,
     private readonly basePath: IBasePath,
-    private readonly repositoryPromise: Promise<ISavedObjectsRepository>
-  ) {}
+    private readonly repositoryPromise: Promise<ISavedObjectsRepository>,
+    stop$: Subject<void>,
+    bufferTimeMs: number = 10_000
+  ) {
+    this.fieldsToIncrement$
+      .pipe(
+        takeUntil(stop$),
+        // Buffer until either the timer or a forced flush occur
+        bufferWhen(() => merge(timer(bufferTimeMs, bufferTimeMs), this.flush$)),
```

Comment:
Perhaps `interval` is slightly preferable here?
```suggestion
        bufferWhen(() => merge(interval(bufferTimeMs), this.flush$)),
```

Reply:
nitnit: not familiar with how many events might be coming in, do you think it is worth defensively adding an upper limit to this buffer size?

Reply:
It looks like `interval` internally uses `timer`: https://github.com/ReactiveX/rxjs/blob/2587ee852eb43eeb0883a7787bac2944d823392b/packages/rxjs/src/internal/observable/interval.ts#L50-L57

Changing it anyways because I agree that it might help with readability.

Reply:
> nitnit: not familiar with how many events might be coming in, do you think it is worth defensively adding an upper limit to this buffer size?

This is a good point! Let me add a max counter somehow.

Reply:
* Use of `interval` fixed in [`274281a` (#180405)](https://github.com/elastic/kibana/pull/180405/commits/274281ade9119a928590d151c652a42ae6a83457)
* Max buffer size introduced in [`41dc22c` (#180405)](https://github.com/elastic/kibana/pull/180405/commits/41dc22c88d38d0eb8cff0cc02c4d7860de7312bb)